modules/solid_mechanics/include/neml2/utils/NEML2Utils.h

```diff
@@ -99,42 +113,41 @@ fromBlob(const std::vector<T> & data)
 {
   // The const_cast is fine because torch works with non-const ptr so that it can optionally handle
   // deallocation. But we are not going to let torch do that.
-  const auto torch_tensor =
-      torch::from_blob(const_cast<T *>(data.data()),
-                       neml2::utils::add_shapes(data.size(), Layout<T>::shape),
-                       torch::TensorOptions().dtype(torch::kFloat64));
+  const auto torch_tensor = at::from_blob(const_cast<T *>(data.data()),
+                                          neml2::utils::add_shapes(data.size(), Layout<T>::shape),
+                                          at::TensorOptions().dtype(neml2::kFloat64));
   return neml2::Tensor(torch_tensor, 1);
 }
 
 /**
- * @brief Directly copy a contiguous chunk of memory of a torch::Tensor to a MOOSE data of type T
+ * @brief Directly copy a contiguous chunk of memory of a at::Tensor to a MOOSE data of type T
  *
- * This assumes the torch::Tensor and T have the same layout, for example both row-major
+ * This assumes the at::Tensor and T have the same layout, for example both row-major
  * with T = RankTwoTensor. If the layouts are different, we may need to reshape/reorder/transpose
- * the torch::Tensor before copying.
+ * the at::Tensor before copying.
```

Comment:
I think you briefly mentioned this, but why the switch here? I thought `at::Tensor` is not differentiable, while `torch::Tensor` is. Has this changed?

Reply:
That does _net_ necessarily mean that. If there was a separate declaration of `torch::Tensor` that would hide `at::Tensor` (see the A::f lookup example in  https://en.cppreference.com/w/cpp/language/qualified_lookup#Namespace_members )

Reply:
Yeah, thanks for pointing that out. I was actually aware of the lookup rule. ATen/core/TensorBody.h is the only declaration of at::Tensor. There's no other declaration under other namespaces that hides it.